### ISSUE TITLE
Some of the tiles graphics from mods failing to load properly after changing world saves

### ISSUE THREAD

**Issue by CV514** [cataclysmbnteam/Cataclysm-BN#1264]
# Describe the bug

Basically, anything that is not core game have a chance not to show proper tile from tileset and use symbol instead in special circumstances.
It's very inconsistent at first glance, but I think I've managed to at least reproduce it to a degree, see step 8 as a conflicting point.

# Steps To Reproduce

This may sound stupid but see context below

1. Create new world with no mods (or some mods, it doesn't matter)
2. Create any character, enter the game
3. Save and quit to menu. Do not exit the game
4. Create another world. This could be done in parallel at step 1. This time - make it only Bright Nights + Aftershock
5. Create any character, enter the game in this new world
6. Debug spawn `Haunted Forest Walker` and/or `Brain Blaster` monsters
7. See that they are missing corresponding graphics data
8. Try to spawn some items from this mod (`energy saber` for reproduce consistency) and see that they are displaying properly, unlike monsters above
9. Be confused

Bonus:
1. Restart the game, load any of the two words
2. See that everything is fine now
3. Save and quit to menu, load another world
4. See that graphics are missing and it seems like only first loaded world doing so properly

# Expected behavior

RTX ON. No confusion.

# Screenshots

![изображение](https://user-images.githubusercontent.com/60776130/149513108-1c61cbbe-4921-44b3-988a-466c14fb2f0a.png)

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 43e3981 [64-bit]
- Graphics Version: Tiles [UNDEAD_PEOPLE] (NEW VERSION)
- Game Language: English [en_US]
- Mods loaded: [
    Bright Nights [dda],
    Aftershock [aftershock]
]
# Additional context

This can be temporary fixed in three ways:
1. Go into options, change tileset to something else from the current one, save changes, revert it back - here, fixed for now
2. Restart the game every time you want to change save world
3. Don't do steps mentioned above - and generally speaking, most players should be fine since they probably won't do that kind of thing anyway. 

But when you're into modmaking and especially trying to mess around tilesets, this may get out of hand pretty fast. And if for some reason someone's playing two worlds without exiting game between sessions - they'll stumble upon this sooner or later.

More or less "constant" fix would be running same mods on every world, that way loading between them does not produce that behavior. but then what's the point of several worlds?

**Comment by olanti-p** [comment]
What happens is the game tries to be helpful and optimize out repeated loadings of same tileset.

If it detects that a tileset has already been loaded, and its id matches id of the required tileset, the loading is skipped, and the old instance is used instead. There are **no** checks done on whether the tileset has been edited, or whether it's loaded with different list of mod tilesets, or whether some of them has been edited as well.

Solution I see:
1. Make the game also check that list of mod tilesets being loaded matches list of mod tilesets in the old instance
2. (Optionally) Add a debug option to disable this optimization completely, for convenience of content creators.

The second one is somewhat redundant as there is already a "Reload tileset" keybind available from main screen which reloads active tileset, all active mod tilesets, and also prints coverage report to `debug.log`.
<details>
    <summary>But that's a rather obscure feature.</summary>

![image](https://user-images.githubusercontent.com/60584843/152704009-f9da0898-20dd-4bd4-83c1-7d7279bdee61.png)
</details>
